### ISSUE TITLE
TLS Reconnect tests flapping

### ISSUE THREAD

**Issue by wallyqs** [nats-io/asyncio-nats#43]
```
=== RUN ClientTLSReconnectTest.test_tls_reconnect
Fatal write error on socket transport
protocol: <asyncio.sslproto.SSLProtocol object at 0x7f09c89760b8>
transport: <_SelectorSocketTransport fd=11 read=polling write=<idle, bufsize=0>>
Traceback (most recent call last):
  File "/opt/python/3.5.1/lib/python3.5/asyncio/selector_events.py", line 702, in write
    n = self._sock.send(data)
OSError: [Errno 9] Bad file descriptor
ERROR: test_tls_reconnect (tests.client_test.ClientTLSReconnectTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/nats-io/asyncio-nats/nats/aio/client.py", line 424, in timed_request
    msg = yield from asyncio.wait_for(future, timeout, loop=self._loop)
  File "/opt/python/3.5.1/lib/python3.5/asyncio/tasks.py", line 390, in wait_for
    raise futures.TimeoutError()
concurrent.futures._base.TimeoutError
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/build/nats-io/asyncio-nats/tests/utils.py", line 295, in wrapper
    loop=test_case.loop))
  File "/opt/python/3.5.1/lib/python3.5/asyncio/base_events.py", line 337, in run_until_complete
    return future.result()
  File "/opt/python/3.5.1/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/opt/python/3.5.1/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/opt/python/3.5.1/lib/python3.5/asyncio/tasks.py", line 386, in wait_for
    return fut.result()
  File "/opt/python/3.5.1/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/opt/python/3.5.1/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/home/travis/build/nats-io/asyncio-nats/tests/client_test.py", line 1050, in test_tls_reconnect
    response = yield from nc.timed_request("example", b'Help!', timeout=1)
  File "/home/travis/build/nats-io/asyncio-nats/nats/aio/client.py", line 428, in timed_request
    raise ErrTimeout
nats.aio.errors.ErrTimeout: nats: Timeout
```

**Comment by oblalex** [comment]
I confirm the issue. It's not about tests. The reason is that `_SelectorTransport` objects for bare socket are being rewritten by `_SelectorSslTransport` objects here: https://github.com/nats-io/asyncio-nats/blob/e1103c40b80487fb248325f3b05d2e1ac26d9aeb/nats/aio/client.py#L867-L874

This leaves initial objects without references to them and garbage collector deletes initial transport objects. For Python 3.4+ this causes underlying socket to get closed: https://github.com/python/cpython/blob/3.6/Lib/asyncio/selector_events.py#L623-L631

This may make `write` or `select` calls to explode with termination of loop's thread.

This is a very tricky and critical bug. It was 100% reproducible on Windows VM, and only ~0.01% on Mac OS. It was also  ~0.01% reproducible on Windows VM with event's loop debug mode turned on. I was hunting the bug around 2 weeks and it has almost driven me crazy.

The simplest solution is just to keep references to initial transport objects. I'll try to come up with a better solution and will make a PR as soon as possible.

**Comment by wallyqs** [comment]
Thanks a lot for the patch and the root cause investigation, I'll take a look and try to cut a release asap.

**Comment by wallyqs** [comment]
Fixed via #49 https://github.com/nats-io/asyncio-nats/pull/49

**Comment by geoai777** [comment]
Am I that lucky to revive 5 year old bug? 
It happens when I try to use `tls` via `ssl context`.

```python
[base_events.py:1707 - default_exception_handler() ] Fatal error on SSL transport
protocol: <asyncio.sslproto.SSLProtocol object at 0x7f47150d7a60>
transport: <_SelectorSocketTransport closing fd=6>
Traceback (most recent call last):
  File "/usr/lib/python3.8/asyncio/selector_events.py", line 910, in write
    n = self._sock.send(data)
OSError: [Errno 9] Bad file descriptor

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/asyncio/sslproto.py", line 685, in _process_write_backlog
    self._transport.write(chunk)
  File "/usr/lib/python3.8/asyncio/selector_events.py", line 916, in write
    self._fatal_error(exc, 'Fatal write error on socket transport')
  File "/usr/lib/python3.8/asyncio/selector_events.py", line 711, in _fatal_error
    self._force_close(exc)
  File "/usr/lib/python3.8/asyncio/selector_events.py", line 723, in _force_close
    self._loop.call_soon(self._call_connection_lost, exc)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 719, in call_soon
    self._check_closed()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 508, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
```
